pkg/compactor/bucket_compactor.go

```diff
@@ -994,19 +997,31 @@ func (c *BucketCompactor) filterOwnJobs(jobs []*Job) ([]*Job, error) {
 	return jobs, nil
 }
 
-// filterJobsByWaitPeriod filters out jobs for which the configured wait period hasn't been honored yet.
-func (c *BucketCompactor) filterJobsByWaitPeriod(ctx context.Context, jobs []*Job) []*Job {
+// filterFirstLevelJobs filters out jobs for which the configured compaction delay or wait period have not yet elapsed.
+// When a compactionDelay is configured, this setting will be used, else the waitPeriod will be used.
+func (c *BucketCompactor) filterFirstLevelJobs(ctx context.Context, jobs []*Job) []*Job {
 	for i := 0; i < len(jobs); {
-		if elapsed, notElapsedBlock, err := jobWaitPeriodElapsed(ctx, jobs[i], c.waitPeriod, c.bkt); err != nil {
-			level.Warn(c.logger).Log("msg", "not enforcing compaction wait period because the check if compaction job contains recently uploaded blocks has failed", "groupKey", jobs[i].Key(), "err", err)
-
-			// Keep the job.
-			i++
-		} else if !elapsed {
-			level.Info(c.logger).Log("msg", "skipping compaction job because blocks in this job were uploaded too recently (within wait period)", "groupKey", jobs[i].Key(), "waitPeriodNotElapsedFor", notElapsedBlock.String())
-			jobs = append(jobs[:i], jobs[i+1:]...)
+		// Filter jobs using compaction delay when enabled
+		if c.compactionDelay > 0 {
+			if jobs[i].MinCompactionLevel() == 1 && jobs[i].MaxTime() > time.Now().Add(-c.compactionDelay).UnixMilli() {
```

Comment:
> jobs[i].MaxTime() > time.Now().Add(-c.compactionDelay).UnixMilli()

This won't work as expected because blocks compacted by ingesters have samples at least 1h old.

Reply:
`compactionDelay` is a new setting that would be configured to something like 90 minutes, giving time for head compaction + upload to occur.